docs/integrations/test-explorer.md

```diff
@@ -0,0 +1,72 @@
+---
+id: test-explorer
+title: Test Explorer
+---
+
+Test Explorer is being under active development, this page explain how it works under the hood.
+
+## Test Discovery
+
+Test Explorer provides two ways to interact with it:
+
+- first one is based on notifications being sent to the client and doesn't require from client to start any actions. All necessary updates are being immediately sent to it.
+
+![](../diagrams/out/test-explorer-server-notifications.png)
+
+- client can also manually ask for test suites from the whole workspace, or for test cases for a given file.
+
+![](../diagrams/out/test-explorer-client-queries.png)
+
+## How it works under the hood
+
+Test Explorer uses BSP's [debug request](https://build-server-protocol.github.io/docs/specification.html#debug-request) to run and debug test classes. TE reuses already available debug support in Metals and build servers such as Bloop or SBT.
+Both run and debug under the hood use BSP's debug request.
+
+### Running and debugging
+
+Following diagram ([source](https://github.com/scalacenter/bloop/blob/12bdc7f97cc3970d3e22a8b513f4b609c813f0a7/docs/assets/dap-example-metals.png)) represents what communication between language client, language server and build server. (It's very likely that in the future Debug Adapter will be inlined in the Metals to reduce number of moving parts).
```

Comment:
> in the future Debug Adapter will be inlined in the Metals to reduce number of moving parts.

Really?

Reply:
Hmm, people on the streets talk too much :P